rep-2009.rst

```diff
@@ -0,0 +1,468 @@
+  The exact function used to determine how to use these priorities can be overwritten by the user.
+  In general, higher numbers indicate higher priority, zero indicates no preference, and negative numbers indicate a vote against using a supported message type.
+
+:Selected Message Type:
+  A supported type that has been selected for use during negotiation.
+
+:Negotiating Publisher:
+  A publisher with preferences as to which supported message type it prefers to publish.
+  It can publish one or more selected message types on the same number of topics.
+
+:Negotiating Subscriber:
+  A subscriber with preferences as to which supported message type it prefers to subscribe to.
+  It subscribes to a single selected message type on a single topic.
+
+:Negotiating Publisher and Subscriber Pair:
+  A singe node that has pair of negotiating publishers and subscribers that are related to each other in that the subscribers preferences may be based on the selected message type of the publisher.
+  An example would be if the subscriber receives data, modifies it, and then publishes it.
+  In this case, it is desirable to have the publisher and subscriber use the same selected message type.
+  However, the selected message type may not be known before the publisher negotiates it.
+
+Specification
+=============
+
+High-level Goals
+----------------
+
+The proposed change has a few goals:
+
+- Provide a way for nodes to publish different types of messages.
+- Support publishing in multiple formats at the same time.
+- Only necessary publishers and subscribers are active at any given time.
+- A publisher or subscriber can delay stating its preferences to wait for additional information.
+- Core functions are exposed to the user to overwrite, namely the ability to choose how priorities are used to pick the selected message types and the ability to delay stating preferences.
+
+
+Defining Supported Message Types
+--------------------------------
+
+For each supported message type, the following information must be known:
+
+- The topic that the supported message type should be published on.
+- The message type that will be sent (either a ROS 2 message type, such as  ``sensor_msgs::msg::Image``, or a custom data type).
+- The relative priority of the supported message type.
+
+The message type is important because multiple supported message types may be sent with the same ROS 2 message type.
+For example, a publisher may publish image data in the format of YUV420 or ARGB8888, both of which could be sent in a ``sensor_msgs::msg::Image``.
+
+Once there is a set of supported message types, negotiating publishers and subscribers can negotiate which type they prefer.
+
+
+Defining Negotiating Publishers and Subscribers
+-----------------------------------------------
+
+Negotiating publishers and subscribers both require a list of supported message types and a service topic that will be used to negotiate the selected message types.
```

Comment:
I'd avoid using the term "service topic" here, as those terms are very overloaded in ROS.  Maybe "base topic"?

Reply:
Fixed in https://github.com/ros-infrastructure/rep/pull/325/commits/c88fc7e0ffe729a3a3699d6b8c0883e677cf78d3.